### PR TITLE
chore(ci): int test yaml file detection

### PR DESCRIPTION
**Pull request by neuronull**
- Adds file detection for changes to the yaml files used in the creation of the int test containers.

### CODE REVIEW

File: .github/workflows/changes.yml [hunk 1]
```diff
@@ -225,116 +227,152 @@ jobs:
...
             - "src/sinks/datadog_archives.rs"
             - "src/sinks/util/**"
+            - "scripts/integration/datadog-agent/**"
+            - "scripts/integration/datadog-logs/**"
+            - "scripts/integration/datadog-metrics/**"
+            - "scripts/integration/datadog-traces/**"
+          dnstap:
+            - "src/internal_events/dnstap.rs"
+            - "src/sources/dnstap/**"
+            - "scripts/integration/dnstap/**"
```
Comment (neuronull) [inline .github/workflows/changes.yml:261]: 👀

Comment (spencergilbert) [reply .github/workflows/changes.yml:261]: 👀 were we not triggering `dnstap` tests individually?

**Review by spencergilbert — APPROVED** [review]
I don't love the duplication, but don't have any suggestions to improve that.

File: .github/workflows/changes.yml (hunk 1 above)
Comment (neuronull) [reply .github/workflows/changes.yml:261]: The dnstap tests were getting run as part of the merge queue checks, but not in PRs based on file detection. Probably went unnoticed.

**Pull request merged**